pymc3/distributions/multivariate.py

```diff
@@ -255,56 +262,53 @@ def random(self, point=None, size=None):
         -------
         array
         """
-        if size is None:
-            size = tuple()
+        size = to_tuple(size)
+
+        param_attribute = "chol_cov" if self._cov_type == "chol" else self._cov_type
+        mu, param = draw_values([self.mu, getattr(self, param_attribute)], point=point, size=size)
+        if tuple(self.shape):
+            dist_shape = tuple(self.shape)
         else:
-            if not isinstance(size, tuple):
-                try:
-                    size = tuple(size)
-                except TypeError:
-                    size = (size,)
+            if is_fast_drawable(self.mu) or point:
```

Comment:
Why do you need to test if `self.mu` is fast drawable? You already have concrete values drawn from `self.mu` in the `mu` variable

Reply:
If the parameters to distributions are fast drawable, then they are returned as it is from `draw_values` function. And they do not come prepended with sample shape.
But if these parameters are pymc3 variables or theano variables consisting of pymc3 variables, I observed that their samples have sample shape as their first dimensions.

Earlier we were doing something like `mu.shape[:len(size)] == size`, this condition can hold true for fast drawable elements but represents wrong intention. Because let's say, mu is of `batch shape=3` and we ask for `sample shape=3`, then drawing values and inferring something from above condition is wrong.

I am not fully sure if doing so, handles all possible ways of defining variables in PyMC3. So, I mentioned to see this PR as a potential fix.